lib/user.rb

```diff
@@ -0,0 +1,27 @@
+class User
+    attr_reader :name,
+                :jokes
+
+  def initialize(name)
+    @name = name
+    @jokes = []
+  end
+
+  def learn(joke)
+    @jokes << joke
+  end
+
+  def tell(name, joke)
+    name.jokes << joke
+  end
+
+#Couldnt figure out the iteration
```

Comment:
I think you're on the right track here, but maybe using not quite the right tool.  I think `each` or `find` would be better in this scenario:

```ruby
#using each
found_joke = nil
@jokes.each do |joke|
  if joke.id == number
    found_joke = joke
  end
end
found_joke

#using find
@jokes.find do |joke|
  joke.id == number
end
```